lib/Dialect/Moore/Transforms/MooreDeclarations.cpp

```diff
@@ -0,0 +1,79 @@
+//===- MooreDeclarations.cpp - Collect net/variable declarations ----------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines the MooreDeclarations pass.
+// Use to collect net/variable declarations and bound a value to them.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetail.h"
+#include "circt/Dialect/Moore/MoorePasses.h"
+
+using namespace circt;
+using namespace moore;
+
+namespace {
+struct MooreDeclarationsPass
+    : public MooreDeclarationsBase<MooreDeclarationsPass> {
+  void runOnOperation() override;
+};
+} // namespace
+
+void Declaration::addValue(Operation *op) {
+  TypeSwitch<Operation *, void>(op)
+      .Case<VariableOp>([&](auto op) {
+        auto value = op.getInitial();
```

Comment:
Maybe a variable declaration doesn't have an initial value(`reg q;`), but when it is used in sequential logic
(`always@(posedge clk) begin if(rst) q <= 1'b0; else q <= 1'b1; end  assign out = q;`). 
This way of writing is legal. However, `hw.wire` must have a value as its operand. Should we create a default value(0, type is the same as req q;) for it? And I think this conforms to the IEEE Standard of SV. If not, when `clk` keeps a low electrical level, `q` doesn't have a value, and `assign out = q` is also meaningless.

Reply:
I have another idea about this. Maybe not having an initial value is permissive. We can call `replaceAllUsesWith()` to update the def-use chain. And this can avoid producing a magic name.

Reply:
Yeah I think you can pick an initial value here. According to the spec, variables and ports without any assignment get the default value of their corresponding type, which for `bit` is `0` and for `logic` is `X`. So feel free to pick either one here.

Reply:
Get it!